[changed] Remove extraneous utils/Object.assign.js

AlexKVal opened the pull request:
By using https://github.com/JedWatson/babel-plugin-object-assign

We get rid of extraneous `utils/Object.assign.js` and even decrease the size of distr files :smile: 
from 233,539 to 231,746 not minified -207 bytes
from 113,874 to 113,451 minified -423 bytes

As for now Babel places polifill for `Object.assign` in generated ES5 code anyway.
```js
var _extends = Object.assign || function (target) {
```

After this commit Babel would place it also into `Interpolate.js` and `OverlayTrigger.js`
instead of requiring it from `utils/Object.assign.js`.
:tada:

mtscout6 commented on .babelrc:
```diff
@@ -1,5 +1,8 @@
     "es7.objectRestSpread"
+  ],
+  "plugins": [
+    "babel-plugin-object-assign"
```
Would we need this if we were using [`babel-runtime`](http://babeljs.io/docs/usage/runtime/)?

mtscout6 replied:
According to this documentation: https://www.npmjs.com/package/babel-plugin-object-assign

Can you just declare:

```
"plugins": [
  "object-assign"
]
```

mtscout6 commented on package.json:
```diff
@@ -45,6 +45,7 @@
     "babel-core": "^5.1.10",
     "babel-eslint": "^3.0.1",
     "babel-loader": "^5.0.0",
+    "babel-plugin-object-assign": "^1.1.0",
```
Likewise with first comment about babel-runtime? Also, how will this fly with the bower release?

AlexKVal commented:
> with the bower release?

:+1:  I've checked generated ES5 code. No changes.
Just removed additional module `utils/Object.assign` 
and added one-liner-polifill into `Interpolate.js` and `OverlayTrigger.js`

As for `babel-runtime`, I'll check it.

mtscout6 commented:
You know what I updated #693, as I think that will be a requirement for the bower-runtime usage.

AlexKVal commented:
As you already know it, we cannot use `babel-runtime` as is for now because of `Bower/AMD`.


We can use it after #693 and it will result in following:
`.eslint`
```js
{
  "optional": [
    "es7.objectRestSpread",
    "runtime"
  ]
}
```
Doesn't need `babel-plugin-object-assign`.

But sizes.. hmm.. it's interesting.. they will be worse than now:
233,539 | 238,788 +5249 bytes
113,874 | 116,325 +2451 bytes

AlexKVal commented:
`babel-runtime` adds these three lines into every file in `lib`
(and almost the same into `amd/**`)
```js
var _Object$defineProperty = require('babel-runtime/core-js/object/define-property')['default'];

var _Object$assign = require('babel-runtime/core-js/object/assign')['default'];

var _interopRequireDefault = require('babel-runtime/helpers/interop-require-default')['default'];
```

mtscout6 commented:
Really wow I would not have expected it to be bigger... but that's a discussion for another issue.

AlexKVal commented:
Yeah. Neither was I.

AlexKVal commented:
As for the trimmed bower after #693.

We then could add just this one line into those two files: `Interpolate.js` and `OverlayTrigger.js`
```js
require('babel-core/polyfill');
```

And that's all. :tada: 

I've checked it. :sparkle:

mtscout6 commented:
And https://github.com/react-bootstrap/react-bootstrap/pull/751#discussion_r31264249

AlexKVal commented:
Done.

mtscout6 commented:
LGTM

AlexKVal commented:
Updated #693 
added `Use require('babel-core/polyfill') in Interpolate.js and OverlayTrigger.js and remove babel-plugin-object-assign dependency.` line to not forget to do it.
